us_imputation_benchmarking/models/imputer.py

```diff
@@ -0,0 +1,88 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Optional, Set, Union
+
+import numpy as np
+import pandas as pd
+
+from us_imputation_benchmarking.utils.logging_utils import get_logger
+
+
+class Imputer(ABC):
+    """
+    Abstract base class for imputation models.
+
+    All imputation models should inherit from this class and implement
+    the required methods.
+    """
+
+    def __init__(self, random_seed=None) -> None:
+        """Initialize the imputer model."""
+        self.predictors: Optional[List[str]] = None
+        self.imputed_variables: Optional[List[str]] = None
+        self.seed = random_seed
+        self.logger = get_logger(
```

Comment:
We should move to using just the standard `logging` library, so for example in each module where you want to log things:

```python
self.logger = logging.getLogger(__name__ etc)
self.logger.debug("Debugging log message!")
```